torchrec_dlrm/dlrm_main.py

```diff
@@ -298,6 +298,11 @@ def parse_args(argv: List[str]) -> argparse.Namespace:
         action="store_true",
         help="Print learning rate every iteration.",
     )
+    parser.add_argument(
+        "--allow_tf32",
+        action="store_true",
```

Comment:
do we want to default to True?

I'm not sure why we (pytorch) defaults to False here: https://pytorch.org/docs/stable/notes/cuda.html#tensorfloat-32-tf32-on-ampere-devices cc @ngimel

Reply:
I think in general this could be hardcoded to `True`. Definitely it will be useful to run many training jobs as one will get results earlier (for example, for parameter search).

On the other hand, for the final [MLPerf convergence statistics](https://github.com/mlcommons/logging/blob/master/mlperf_logging/rcp_checker/training_2.1.0/rcps_dlrm.json) we may use it as long as everyone involved agrees to do so